src/couchdb/setup-database.ts

```diff
@@ -0,0 +1,56 @@
+import nano, { DatabaseCreateParams, DocumentScope } from 'nano'
+
+import { matchJson } from '../util/match-json'
+
+/**
+ * Describes a single Couch database that should exist.
+ */
+export interface DatabaseSetup {
+  name: string
+  options?: DatabaseCreateParams
+
+  // Documents that should exactly match:
+  documents?: { [id: string]: object }
+
+  // Documents that we should create, unless they already exist:
+  templates?: { [id: string]: object }
+}
+
+/**
+ * Ensures that the requested database exists in Couch.
+ */
+export async function setupDatabase(
+  couchUri: string,
+  setupInfo: DatabaseSetup
```

Comment:
Can you add param like in my PR: `opt: SetupCouchDbOptions = { log: console.log }`

```ts
export interface SetupCouchDbOptions {
  log: ((...args: any) => void) | null
}
```

Reply:
I did it with tighter types. If you want silence, pass `()=>{}` instead of `null`. I also made it take just a `string`, since that's all we need, and you can always pass something looser like `(...rest: any[]) => void` if you have one.